test/button/button-subtle-perceptual.js

```diff
@@ -0,0 +1,72 @@
+const puppeteer = require('puppeteer');
+const visualDiff = require('visual-diff');
+
+visualDiff.run((ctx) => {
+
+	describe('d2l-button-subtle', function() {
+
+		let browser, page;
+
+		before(async() => {
+			browser = await puppeteer.launch();
+			page = await browser.newPage();
+		});
+
+		after(() => browser.close());
+
+		const runTests = () => {
+
+			it('normal', async function() {
+				const rect = await ctx.puppeteer.getRect(page, '#normal');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('mouse', async function() {
+				await page.hover('#normal');
+				const rect = await ctx.puppeteer.getRect(page, '#normal');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('focus', async function() {
+				await page.click('#normal');
+				const rect = await ctx.puppeteer.getRect(page, '#normal');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('disabled', async function() {
+				const rect = await ctx.puppeteer.getRect(page, '#disabled');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+			it('with-icon', async function() {
+				const rect = await ctx.puppeteer.getRect(page, '#with-icon');
+				await ctx.puppeteer.screenshotAndCompare(page, this.test.fullTitle(), { clip: rect });
+			});
+
+		};
+
+		describe('wide', function() {
+
+			beforeEach(async function() {
+				await page.setViewport({width: 800, height: 800, deviceScaleFactor: 2});
+				await page.goto(`${ctx.serverInfo().baseUrl}/test/button/button-subtle-fixture.html`, {waitUntil: ['networkidle2', 'load']});
+			});
+
+			runTests();
+
+		});
+
+		describe('narrow', function() {
+
+			beforeEach(async function() {
+				await page.setViewport({width: 600, height: 800, deviceScaleFactor: 2});
+				await page.goto(`${ctx.serverInfo().baseUrl}/test/button/button-subtle-fixture.html`, {waitUntil: ['networkidle2', 'load']});
```

Comment:
FYI: I dug into `polyserve` a bit and was able to extract all the info needed to provide a `baseUrl`.  Basically gets the Url info from `polyserve` in the same way that `polymer serve` does when it writes out to the console.  This should make these navigation statements a little nicer.